llvm/lib/CodeGen/InterleavedAccessPass.cpp

```diff
@@ -639,7 +705,8 @@ bool InterleavedAccessImpl::lowerDeinterleaveIntrinsic(
       return false;
 
     // Check mask operand. Handle both all-true/false and interleaved mask.
-    Mask = getMask(getMaskOperand(II), Factor, getDeinterleavedVectorType(DI));
+    std::tie(Mask, std::ignore) =
+        getMask(getMaskOperand(II), Factor, getDeinterleavedVectorType(DI));
```

Comment:
Should we be asserting here that the mask factor returned == factor?

Reply:
yeah good point, it's fixed now.